### PR TITLE
Epic 7.7.1 imp

### PR DESCRIPTION
**Pull request by TekuriSaiAkhil**
1.  Error Management
2. Unit Tests
3. Code reorganizing

### CODE REVIEW

File: app/features/syllabus_generator/core.py [hunk 1]
```diff
@@ -21,26 +21,32 @@ def executor(grade_level: str,
...
+                                assessment_methods,
+                                grading_scale,
+                                summary)
+        
+        syllabus = generate_syllabus(request_args, verbose= verbose)
+
+    except Exception as e:
+        error_message = f"Error in executor: {e}"
+        logger.error(error_message)
+        raise ValueError(error_message)
```
Comment (AaronSosaRamos) [inline app/features/syllabus_generator/core.py]: This approach is fine. Could you please create specific error classes for specify the error that could be catched here?

**Comment by AaronSosaRamos** [comment]
@TekuriSaiAkhil Could you please add some screenshots when you run your unit tests please?

**Comment by TekuriSaiAkhil** [comment]
Unit Tests
Screenshot 2024-07-11 at 12 33 08 PM

**Review by AaronSosaRamos — APPROVED** [review]
LGTM! Good work

**Pull request merged**